pkg/cloudprovider/cloudprovider.go

```diff
@@ -169,6 +169,11 @@ func (c *CloudProvider) createVMInstance(ctx context.Context, nodeClass *v1beta1
 	if err != nil {
 		return nil, err
 	}
+	// Propagate single-value wellKnownLabels from the nodeClaim requirements to the labels,
+	// as this is required for scheduling in core to work correctly. If this is not done, on subsequent scheduling passes before the Node is
+	// registered, the NodeClaim will not have the labels required to match the Pod and so a new NodeClaim will be created each time.
+	newNodeClaim.Labels = lo.Assign(newNodeClaim.Labels, labelspkg.GetWellKnownSingleValuedRequirementLabels(scheduling.NewNodeSelectorRequirementsWithMinValues(nodeClaim.Spec.Requirements...)))
```

Comment:
Do we understand why AWS provider gets away with not doing this?

Reply:
I could also smuggle these other items through tags but I'm not sure it's needed, WDYT?

Either way, I'll update the comment here a bit to make this more clear.

Reply:
No, I think the way you are doing it now is fine. Good to capture the rationale, this helps! (And makes it easier to revisit if we need to)